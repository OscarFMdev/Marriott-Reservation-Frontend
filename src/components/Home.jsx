import React from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import img from '../assets/img/marriott-wobg.png';
import home from './componentsCss/Home.module.css';

const Home = () => {
  const { error, message } = useSelector((state) => state.auth);

  return (
    <div className={home.background}>
      <img src={img} alt="Logo" className={home.hotel} />
      {message && <p>{error}</p>}
      <h1 className={home.header}>The new Marriott Hotel</h1>
      <Link to="/login">
        <button type="button" className={home.loginBtn}>Log in</button>
      </Link>
      <Link to="/signup">
        <button type="button" className={home.signupBtn}>Sign up</button>
      </Link>
    </div>
  );
};

export default Home;
