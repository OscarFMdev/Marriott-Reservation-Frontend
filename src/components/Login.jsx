import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink, useNavigate } from 'react-router-dom';
import { InputText } from 'primereact/inputtext';
import { Button } from 'primereact/button';
import { Password } from 'primereact/password';
import { signIn, getUser } from '../redux/reducer/Auth/auth';
import stl from './componentsCss/Login.module.css';

const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const { message } = useSelector((state) => state.auth);

  const handleSubmit = (e) => {
    e.preventDefault();
    const user = {
      email,
      password,
    };
    dispatch(signIn(user));
  };

  useEffect(() => {
    if (message === 'Logged in succesfully.') {
      dispatch(getUser());
      navigate('/rooms');
    }
  }, [message, navigate]);

  return (
    <div className="p-grid p-justify-center">
      <div className="p-col-12 p-md-8 p-lg-6">
        <div className={stl.card}>
          { !message ? null : <p className={stl.success}>{message}</p>}
          <h1>Login</h1>
          <form onSubmit={handleSubmit}>
            <div className="p-field pb-2">
              <span className="p-input-icon-right">
                <i className="pi pi-envelope" />
                <InputText
                  id="email"
                  type="email"
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                  data-testid="email"
                  placeholder="Email"
                  required
                />
              </span>
            </div>

            <div className="p-field pb-3">

              <Password
                id="password"
                value={password}
                onChange={(e) => setPassword(e.target.value)}
                data-testid="password"
                placeholder="Password"
                required
                feedback={false}
                toggleMask
              />

            </div>
            <Button type="submit" label="Login" className="bg-green-500 border-200" />
            <p>
              Do not Have an Account?
              <span>
                <NavLink to="/signup"> Sign Up</NavLink>
              </span>
            </p>
          </form>
        </div>
      </div>
    </div>
  );
};

export default Login;
