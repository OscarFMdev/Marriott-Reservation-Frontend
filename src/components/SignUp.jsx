/* eslint-disable camelcase */
import React, { useState, useEffect } from 'react';
import { InputText } from 'primereact/inputtext';
import { Password } from 'primereact/password';
import { Button } from 'primereact/button';
import { NavLink, useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { signUp } from '../redux/reducer/Auth/auth';

const SignUp = () => {
  const navigate = useNavigate();
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [password_confirmation, setPasswordConfirmation] = useState('');
  const dispatch = useDispatch();
  const { status, message } = useSelector((state) => state.auth);

  const handleSubmit = (e) => {
    e.preventDefault();
    const user = {
      name,
      email,
      password,
      password_confirmation,
    };
    dispatch(signUp(user));
  };

  useEffect(() => {
    if (message === 'Sign up success') {
      navigate('/login');
    }
  }, [message, navigate]);

  return (
    <div className="p-grid p-justify-center">
      <div className="p-col-12 p-md-8 p-lg-6">
        <div className="card">
          { status === 'failed' && <p>{message}</p> }
          <h1>Sign Up</h1>
          <form onSubmit={handleSubmit}>
            <div className="p-field pb-2">
              <span className="p-input-icon-right">
                <i className="pi pi-user" />
                <InputText
                  id="name"
                  type="text"
                  value={name}
                  onChange={(e) => setName(e.target.value)}
                  label="Name"
                  placeholder="Full Name"
                  required
                />
              </span>
            </div>

            <div className="p-field pb-2">
              <span className="p-input-icon-right">
                <i className="pi pi-envelope" />
                <InputText
                  id="email"
                  type="email"
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                  label="Email"
                  placeholder="Email"
                  required
                />
              </span>
            </div>

            <div className="p-field pb-2">
              <Password
                id="password"
                value={password}
                onChange={(e) => setPassword(e.target.value)}
                label="password"
                placeholder="Password"
                required
                toggleMask
              />
            </div>

            <div className="p-field pb-2">
              <Password
                id="password_confirmation"
                value={password_confirmation}
                onChange={(e) => setPasswordConfirmation(e.target.value)}
                label="passconf"
                placeholder="Password"
                required
                feedback={false}
                toggleMask
              />
            </div>

            <Button label="Sign Up" type="submit" className="bg-green-500 border-200" />
            <p>
              Already have an account?
              <NavLink to="/login">Login</NavLink>
            </p>
          </form>
        </div>
      </div>
    </div>
  );
};

export default SignUp;
