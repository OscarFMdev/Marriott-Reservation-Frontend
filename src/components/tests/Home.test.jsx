import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Home from '../Home';
import SignUp from '../SignUp';
import Login from '../Login';
import store from '../../redux/configureStore';

describe('It displays the homepage title', () => {
  test('Homepage title', () => {
    render(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    );
    expect(screen.getByText('The new Marriott Hotel')).toBeDefined();
  });
});

describe('Test redirection of the login and signup buttons', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router>
          <Home />
          <Login />
          <SignUp />
        </Router>
      </Provider>,
    );
  });
  test('Log in Redirection', () => {
    fireEvent.click(screen.getByRole('button', { name: 'Log in' }));
    expect(screen.getByText('Do not Have an Account?')).toBeDefined();
  });
  test('Sign Up Redirection', () => {
    fireEvent.click(screen.getByRole('button', { name: 'Sign up' }));
    expect(screen.getByText('Already have an account?')).toBeDefined();
  });
});

test('Home Page renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Home />
      </Router>
      h
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
