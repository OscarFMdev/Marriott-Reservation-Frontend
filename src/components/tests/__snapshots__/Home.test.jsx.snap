// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page renders correctly 1`] = `
[
  <div
    className="_background_ca55bd"
  >
    <img
      alt="Logo"
      className="_hotel_ca55bd"
      src="/src/assets/img/marriott-wobg.png"
    />
    <h1
      className="_header_ca55bd"
    >
      The new Marriott Hotel
    </h1>
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="_loginBtn_ca55bd"
        type="button"
      >
        Log in
      </button>
    </a>
    <a
      href="/signup"
      onClick={[Function]}
    >
      <button
        className="_signupBtn_ca55bd"
        type="button"
      >
        Sign up
      </button>
    </a>
  </div>,
  "h",
]
`;
